compiler/rustc_mir_transform/src/cleanup_post_borrowck.rs

```diff
@@ -25,6 +30,12 @@ impl<'tcx> MirPass<'tcx> for CleanupPostBorrowck {
                 match statement.kind {
                     StatementKind::AscribeUserType(..)
                     | StatementKind::Assign(box (_, Rvalue::Ref(_, BorrowKind::Fake, _)))
+                    | StatementKind::Coverage(box Coverage {
+                        // These kinds of coverage statements are markers inserted during
+                        // MIR building, and are not needed after InstrumentCoverage.
+                        kind: CoverageKind::BlockMarker { .. } | CoverageKind::SpanMarker { .. },
```

Comment:
Please also add this to validation to ensure they never get reintroduced

Reply:
I added an appropriate check to `CfgChecker` [(diff)](https://github.com/rust-lang/rust/compare/313b6a04ed88bcbe2a47d4d9395478ddaeec986b..42d7797d02f5f0021c8904efc5f72789c834ca40).